dev/planning-example.py

```diff
@@ -0,0 +1,292 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# ============================
+# Observation Planning Toolbox
+# ============================
+"""
+The observation planning toolbox consists of a number of entities that
+simplify the planning and scheduling of observations by abstracting much
+of the boilerplate code necessary to accomplish the same thing using
+core astropy or ephemerides objects.  These can be thought of as common
+idioms or convenience classes and functions that would be recreated over
+and over by astronomers and or software engineers building telescope
+scheduling software.
+
+Important assumptions:
+
+- It should be convenient to specify dates and times in the local timezone
+  of the observer.
+
+- It should be convenient to do timezone conversions easily.
+
+  This means methods that can take dates can either take astropy
+  dates (as used internally) or python datetime objects (not naive, but
+  tagged with timezone).  So dates can be passed in as timezone-aware
+  datetime objects and results can be easily converted between such.
+
+"""
+
+# ================
+# Observer objects
+# ================
+"""
+An observer defines a single terrestrial location and applicable
+environmental attributes thereof that affect observation calculations
+at that location.
+"""
+# Define an observer by full specification.
+# Latitude/Longitude take input as strings or quantities,
+# elevation, pressure, temperature must be quantities.
+# Accept `timezone` from pytz
+from astroplan import Observer
+import astropy.units as u
+import pytz
+
+obs = Observer(name='Subaru Telescope',
+               longitude='-155:28:48.900',
+               latitude='+19:49:42.600',
+               elevation=4163 * u.meter,
+               pressure=0.615 * u.bar,
+               relative_humidity=0.11,
+               temperature=0 * u.deg_C,
+               timezone=pytz.timezone('US/Hawaii'),
```

Comment:
How about accepting both `pytz.timezone` and `str` (and internally always storing a `pytz.timezone` object?
Just passing a string is convenient, because then the user doesn't have to import `pytz`, and there's lots of precedence for this pattern in Astropy, see e.g. the [SkyCoord](http://astropy.readthedocs.org/en/latest/api/astropy.coordinates.SkyCoord.html) class which passes strings along to `Unit` and `Time` constructors.

I like this pattern and pass strings all the time, so I'd be +1 to do the same e.g. for `elevation`, `pressure`, ... if the user passes a string it's passed to the Quantity constructor.

(Ignore this comment if this has been discussed and rejected already or if this was how you actually wanted to implemented it.)